File: examples/tentacles/polyglot/echo.php
<?php
$req = json_decode(stream_get_contents(STDIN), true) ?: [];
echo json_encode([
  'v' => 1,
  'id' => $req['id'] ?? null,
  'ok' => true,
  'result' => [
    'arm' => 'echo-php',
    'lang' => 'php',
    'action' => $req['action'] ?? null,
    'echo' => $req['payload'] ?? new stdClass(),
  ],
]);

File: examples/tentacles/polyglot/echo.rs
// Compile: rustc -o echo-rs echo.rs
// Then point tentacle.yaml binary at echo-rs.
use std::io::{self, Read};
fn main() {
    let mut s = String::new();
    io::stdin().read_to_string(&mut s).ok();
    if s.is_empty() { s = "{}".into(); }
    print!("{{\"v\":1,\"ok\":true,\"result\":{{\"arm\":\"echo-rs\",\"lang\":\"rust\"}}}}");
}
